core/src/apps/tezos/helpers.py

```diff
@@ -43,12 +46,22 @@ def validate_full_path(path: list) -> bool:
     Validates derivation path to equal 44'/1729'/a',
     where `a` is an account index from 0 to 1 000 000.
     """
-    if len(path) != 3:
+    length = len(path)
+    if length < 3 or length > 4:
         return False
     if path[0] != 44 | HARDENED:
         return False
     if path[1] != 1729 | HARDENED:
         return False
     if path[2] < HARDENED or path[2] > 1000000 | HARDENED:
         return False
+    if length > 3 and (path[3] < HARDENED or path[3] > 1000000 | HARDENED):
```

Comment:
Including the 4th component to the derivation path in this PR

Reply:
still not too sure about this - you're allowing two distinct components a wide range of values, and also it's a fourth completely separate scheme :(

what paths are used on Ledger?

Reply:
The default path on ledger is m/44'/1729'/0'/0'. See https://github.com/obsidiansystems/ledger-app-tezos/blob/master/README.md#importing-the-key-from-the-ledger-nano-s

Reply:
would it be a problem to fix the 4th path component at `0'`?

Reply:
The idea is, that we want to keep the option to use the paths up to now. e.g:

m/44'/1729'/0'
m/44'/1729'/1'
m/44'/1729'/2' 
etc.

and also allow ledger user to migrate to trezor. for that we need paths, e.g.:

m/44'/1729'/0'/0'
m/44'/1729'/0'/1'
m/44'/1729'/0'/2'
etc.

So fixing the 4th component would be a problem.

Reply:
gah, why does Ledger need to do a completely arbitrary thing yet again :(

in that case let's enforce that
a) either `length == 3` and then `0' <= path[2] <= 1M'`
b) or `length == 4` and then `path[2] == 0'` and `0' <= path[3] <= 1M'`

Reply:
Yeah, that should do the trick :)